src/main/java/org/jboss/set/assist/evaluator/impl/payload/AssociatedPullRequestEvaluator.java

```diff
@@ -92,6 +93,7 @@ public void eval(PayloadEvaluatorContext context, Map<String, Object> data) {
         for (PullRequest pullRequest : relatedPullRequests) {
             CommitStatus commitStatus = PatchHomeService.retrieveCommitStatus(pullRequest);
             boolean mergedInFuture = CommitHomeService.isMergedInFutureBranch(pullRequest);
+            MergedStatus merged = mergedInFuture ? MergedStatus.MERGED_IN_FUTURE : (pullRequest.isMerged() ? MergedStatus.MERGED : MergedStatus.UNMERGED);
```

Comment:
@michpetrov Could you add a little explanation here? Why this check `mergedInFuture` first ? and then conclude it's merged if commits in PR found in future branch ? 

It's not "real" merged to me in that case, unless you want to assume that commits in PR found in future branch must be included in the real product branch later.

Reply:
My understanding of the lifecycle is this:
* you open a PR to e.g. `7.3.x` (UNMERGED)
* PR is merged (MERGED)
* at some point `7.3.x-future` is updated with the contents of `7.3.x` (MERGED_IN_FUTURE)

It's not possible for a PR to be merged into `7.3.x-future` and not be merged into `7.3.x`, is it?

Reply:
AFAIU, the lifecycle with `future branch` involved is a bit different, the propose of that short-lived branch is described here: [Future branch](https://docs.google.com/document/d/14GL7YbjjCQRyHEMri06adQ0zummlR1jLCyrmHdYVA1E/edit#heading=h.6ztx6pyrotyq).

Sometimes, it's not necessary to use that branch e.g. https://github.com/jbossas/jboss-eap7/tree/7.3.x-future (it seems that it's unused in EAP 7.3.6 CP development)

Yes, I can't think why a PR merged into `7.3.x-future` and not be merged into `7.3.x` later on, but I just want to confirm with you the `mergedInFuture` checking is on purpose here.